tests/unit_tests/test_matchmaker_algorithm_bucket_teams.py

```diff
@@ -0,0 +1,312 @@
+import math
+import random
+
+import pytest
+from hypothesis import given
+from hypothesis import strategies as st
+
+from server import config
+from server.matchmaker import Search, algorithm
+from server.matchmaker.algorithm.bucket_teams import (
+    BucketTeamMatchmaker,
+    _make_teams,
+    _make_teams_from_single
+)
+from server.rating import RatingType
+
+from .strategies import st_searches_list
+
+
+@pytest.fixture(scope="module")
+def player_factory(player_factory):
+    def make(
+        mean: int = 1500,
+        deviation: int = 500,
+        ladder_games: int = config.NEWBIE_MIN_GAMES + 1,
+        name=None,
+    ):
+        """Make a player with the given ratings"""
+        player = player_factory(
+            ladder_rating=(mean, deviation),
+            ladder_games=ladder_games,
+            login=name,
+            lobby_connection_spec=None,
+        )
+        return player
+
+    return make
+
+
+@pytest.mark.parametrize("make_teams_func", (_make_teams, _make_teams_from_single))
+@given(
+    searches=st_searches_list(max_players=1),
+    size=st.integers(min_value=1, max_value=10),
+)
+def test_make_teams_single_correct_size(searches, size, make_teams_func):
+    matched, _ = make_teams_func(searches, size)
+
```

Comment:
We could `assume(matched != [])` here to make sure the examples being generated are actually relevant. This will also increase the time it takes for the test to finish since it will be generating more examples.